test/parallel/test-require-resolve.js

```diff
@@ -80,3 +80,16 @@ require(fixtures.path('resolve-paths', 'default', 'verify-paths.js'));
     assert.strictEqual(resolvedPaths.includes('/node_modules'), false);
   });
 }
+
+{
+  assert.strictEqual(require.resolve('node:test'), 'node:test');
+  assert.strictEqual(require.resolve('node:fs'), 'node:fs');
+
+  assert.throws(
+    () => require.resolve('node:unknown'),
+    {
+      code: 'MODULE_NOT_FOUND',
+      message: /^Cannot find module 'node:unknown'/,
```

Comment:
Quote from [the documentation](https://nodejs.org/api/modules.html#requireresolverequest-options):
> If the module can not be found, a MODULE_NOT_FOUND error is thrown.